modules/develop/pages/data-transforms/run-transforms.adoc

```diff
@@ -0,0 +1,254 @@
+```
++
+A successful command generates project files in your current directory:
++
+[.no-copy]
+----
+.
+├── go.mod
+├── go.sum
+├── README.md
+├── transform.go
+└── transform.yaml
+----
++
+The `transform.go` file contains the transform logic, and the `transform.yaml` file configures the project.
++
+TIP: When creating a custom data transform, initialization steps can be done either in `main` (because it's only run once at the start of the package) or in Go's standard predefined `init()` function. Although state can be cached in global variables, Redpanda may restart a Wasm module at any point, which causes the state to be lost.
+
+. Implement your project by adding transform logic to `transform.go`. 
++
+The following examples show some basic transforms. Each example can be copied into the `transform.go` file.
++
+[tabs]
+====
+Identity transform::
++
+--
+```go
+package main
+
+import (
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows the basic usage of the package:
+// This transform does nothing but copy the same data from an
+// input topic to an output topic.
+func main() {
+	// Make sure to register your callback and perform other setup in main
+	redpanda.OnRecordWritten(identityTransform)
+}
+
+// This will be called for each record in the source topic.
+//
+// The output records returned will be written to the destination topic.
+func identityTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	return []redpanda.Record{e.Record()}, nil
+}
+```
+--
+Transcoder transform::
++
+--
+
+```go
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"io"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows a transform that converts CSV inputs into JSON outputs.
+func main() {
+	redpanda.OnRecordWritten(csvToJsonTransform)
+}
+
+type Foo struct {
+	A string `json:"a"`
+	B int    `json:"b"`
+}
+
+func csvToJsonTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	// The input data is a CSV (without a header row) that is the structure of:
+	// key, a, b
+	reader := csv.NewReader(bytes.NewReader(e.Record().Value))
+	// Improve performance by reusing the result slice.
+	reader.ReuseRecord = true
+	output := []redpanda.Record{}
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		if len(row) != 3 {
+			return nil, errors.New("unexpected number of rows")
+		}
+		// Convert the last column into an int
+		b, err := strconv.Atoi(row[2])
+		if err != nil {
+			return nil, err
+		}
+		// Marshal our JSON value
+		f := Foo{
+			A: row[1],
+			B: b,
+		}
+		v, err := json.Marshal(&f)
+		if err != nil {
+			return nil, err
+		}
+		// Add our output record using the first column as the key.
+		output = append(output, redpanda.Record{
+			Key:   []byte(row[0]),
+			Value: v,
+		})
+
+	}
+	return output, nil
+}
+```
+
+--
+Validation filter transform::
```

Comment:
@rockwotj these are good  examples but we really also need an e.g. JSON --> Avro transform to show off the schema registry integration (also more commonly used  than CSV)

Reply:
Yeah the examples where supposed to only contain the standard library, and there is no standard library encoding for Avro.

I can show avro using a library that works in TinyGo.

Reply:
I will followup with a better example after this goes in